WonderBrush/src/tools/PenStroke.cpp

```diff
@@ -383,6 +383,8 @@ PenStroke::_StrokePoint(float x, float y, float pressure, uint8* bits, uint32 bp
 	else
 		pressure = fAlpha.max;
 	uint8 maxAlpha = (uint8)(255.0 * pressure);
+	if (y < 0)
+		return;
```

Comment:
This doesn't seem right. From what I remember, `x` and `y` specify the center of the shape that is to be rendered by the `PenStroke`. The `constrainRect` should contain the necessary clipping. If somehow it is possible to render outside the bitmap buffer, then the bug is certainly not at this place. And if it were, then we should just as well check for `x < 0` and also right and bottom buffer bounds.